crates/node/src/network_server/handler/cluster_ctrl.rs

```diff
@@ -230,6 +230,61 @@ impl ClusterCtrlSvc for ClusterCtrlSvcHandler {
             .map(|_| Response::new(SealAndExtendChainResponse::default()))
             .map_err(|err| Status::internal(err.to_string()))
     }
+
+    async fn find_tail(
+        &self,
+        request: Request<FindTailRequest>,
+    ) -> Result<Response<FindTailResponse>, Status> {
+        let request = request.into_inner();
+        let log_id: LogId = request.log_id.into();
+
+        let logs = self.get_logs().await?;
+        let Some(chain) = logs.chain(&log_id) else {
+            return Err(Status::not_found("No log with given id"));
+        };
+
+        let tail = chain.tail();
```

Comment:
Can we use writeable_loglet's error as a signal to know if the log exists to remove the risk of the changes that can happen in between?